From 9d92b8df115f3feecff9cdb2aec9da1acb21932c Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Wed, 18 Sep 2024 17:46:32 +0100
Subject: [PATCH 2/6] Darwin: Allow for as versions that need '-' for std in.

Recent versions of Xcode as require a dash to read from standard
input.  We can use this on all supported OS versions so make it
unconditional.  Patch from Mark Mentovai.

gcc/ChangeLog:

	* config/darwin.h (AS_NEEDS_DASH_FOR_PIPED_INPUT): New.

Signed-off-by: Iain Sandoe <iain@sandoe.co.uk>
(cherry picked from commit 33ccc1314dcdb0b988a9276ca6b6ce9b07bea21e)
---
 gcc/config/darwin.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git gcc/config/darwin.h gcc/config/darwin.h
index 5fd52f2cc0d2..83e4e2fcf555 100644
--- gcc/config/darwin.h
+++ gcc/config/darwin.h
@@ -653,6 +653,8 @@ extern GTY(()) int darwin_ms_struct;
 #define ASM_OPTIONS "%{v} %{w:-W} %{I*}"
 #endif
 
+#define AS_NEEDS_DASH_FOR_PIPED_INPUT
+
 /* Default Darwin ASM_SPEC, very simple. */
 #define ASM_SPEC \
 "%{static} -arch %(darwin_arch) " \
-- 
2.47.0

